### ISSUE TITLE
Implement HLS

### ISSUE THREAD

**Issue by potyt** [ValdikSS/aceproxy#8]
Any idea if it is possible to get this on a Roku or through Plex (ideally without transcoding for the Roku)?

Cheers

**Comment by AlexRevin** [comment]
I think it is easier to forward video stream to RTMP server capable of generating HLS

**Comment by elwebmaster** [comment]
Aceproxy can't forward video streams to RTMP servers as far as I understand. It really shouldn't be that hard to implement HLS given that aceproxy already relies on VLC. See my any_to_m3u8 script which does the same but with FFMPEG: https://github.com/elwebmaster/any_to_m3u8 . I know it can be done with VLC also, but I don't know the correct syntax.

**Comment by AlexRevin** [comment]
Yeah, that's exactly what I mean. Is it possible to stream from aceproxy to ffmpeg? FFMpeg can output the stream to RTMP server which will serve it in the way it likes.

What I see in code, is that video stream is being forwarded through telnet client to vlc. Can't see a reason for not making it a http stream

**Comment by dimitrovs** [comment]
I am not familiar with aceproxy's code, but VLC and FFMPEG have very similar features. It should be possible to make VLC write out HLS chunks instead of mp4. Otherwise you can just use my script on top of aceproxy, this will have the same result but will involve both VLC & FFMPEG (which I think is unnecessary).

**Comment by nickcarterney** [comment]
Hello,
```

    # transcodecmd['100k'] = 'ffmpeg -i - -c:a copy -b 100k -f mpegts -'
    # transcodecmd['mp2'] = 'ffmpeg -i - -c:a mp2 -c:v mpeg2video -f mpegts -qscale:v 2 -'.split()
    # transcodecmd['mkv'] = 'ffmpeg -i - -c:a copy -c:v copy -f matroska -'.split()
    # transcodecmd['default'] = 'ffmpeg -i - -c:a copy -c:v copy -f mpegts -'.split()
    # ----------------------------------------------------
    # Transcoding configuration for HLS
    # ----------------------------------------------------
    # If you use acestream engine ver >= 3.1.5 and vlcuse=True
    # proxy automaticaly switch to HLS (HTTP Live Streaming) instead of HTTP Progressive Download
    # You can use this settings for audio transcoding. This option applies only for Live-stream
```
What is the hls url I can get ? Like this one:
http://178.62.18.13:8000/pid/8933c0e34bbe9448da63a806b3098d19ee909b8d/stream.mp4

https://github.com/AndreyPavlenko/aceproxy/commit/06f4a00649a75bc3507550eb6d2c39bc496ab26a
